node/subsystem/src/messages.rs

```diff
@@ -573,6 +574,74 @@ impl CollationGenerationMessage {
 	}
 }
 
+/// The result type of [`ApprovalVotingMessage::CheckAndImportAssignment`] request.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum AssignmentCheckResult {
+	/// The vote was accepted and should be propagated onwards.
+	Accepted,
+	/// The vote was valid but duplicate and should not be propagated onwards.
+	AcceptedDuplicate,
+	/// The vote was valid but too far in the future to accept right now.
+	TooFarInFuture,
+	/// The vote was bad and should be ignored, reporting the peer who propagated it.
+	Bad,
```

Comment:
There is likely some equivalent type coming from Rob's code

We signal this only when the VRF criteria fails to verify?  It's just an internal report?

We'll eventually have the possibility of a relay chain equivocation about which we do not know here, unless we include two headers in the assignment.  Not a concern in this iteration though

Reply:
Yeah, I think this is if VRF criteria is bad or the slot number isn't in the VRF signature.